components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -552,11 +552,13 @@ void RewardsServiceImpl::OnCreateWallet(
   std::move(callback).Run(static_cast<int32_t>(result));
 }
 
-void RewardsServiceImpl::AddPrivateObserver(RewardsServicePrivateObserver* observer) {
+void RewardsServiceImpl::AddPrivateObserver(
```

Comment:
where is this used?

Reply:
Seems like never used. @SergeyZhukovsky are we going to use it in future or it is just obsolete?

Reply:
we use it to deliver notifications into a panel as it is not a web ui.

Reply:
there's nothing webui specific about the observers. Just the opposite, they exist to decouple that logic so not sure why you would need a new type of observer here instead of creating an android alternative to ExtensionRewardsNotificationServiceObserver

Reply:
it will be investigated in:
https://github.com/brave/brave-browser/issues/6337